src/olympia/scanners/tests/test_tasks.py

```diff
@@ -612,6 +613,72 @@ def test_does_not_run_when_results_contain_errors(self, yara_compile_mock):
         # The task should always return the results.
         assert received_results == self.results
 
+    @mock.patch('olympia.scanners.tasks.statsd.incr')
+    @override_switch('yara-read-binary', active=True)
+    def test_run_in_binary_mode(self, incr_mock):
+        self.upload = self.get_upload('webextension_with_image.zip')
+
+        assert len(ScannerResult.objects.all()) == 0
+        # This rule will match for all files in the xpi.
```

Comment:
That's not correct, I believe.

Reply:
Oh, right, I had copied and pasted the entire test.

Reply:
e2bfd9c225